eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -143,42 +218,130 @@ object Transactions {
-      case _: InputInfo.TaprootInput => false
-      case InputInfo.SegwitInput(outPoint, txOut, redeemScript) =>
-        val sighash = this.sighash(txOwner, commitmentFormat)
-        val inputIndex = tx.txIn.indexWhere(_.outPoint == outPoint)
-        if (inputIndex >= 0) {
-          val data = Transaction.hashForSigning(tx, inputIndex, redeemScript, sighash, txOut.amount, SIGVERSION_WITNESS_V0)
-          Crypto.verifySignature(data, sig, pubKey)
-        } else {
-          false
+    protected def checkSig(sig: ByteVector64, publicKey: PublicKey, sighash: Int, redeemInfo: RedeemInfo): Boolean = {
+      if (inputIndex >= 0) {
+        redeemInfo match {
+          case redeemInfo: RedeemInfo.SegwitV0 =>
+            val data = Transaction.hashForSigning(tx, inputIndex, redeemInfo.redeemScript, sighash, input.txOut.amount, SIGVERSION_WITNESS_V0)
+            Crypto.verifySignature(data, sig, publicKey)
+          case _: RedeemInfo.TaprootKeyPath => ???
+          case _: RedeemInfo.TaprootScriptPath => ???
         }
+      } else {
+        false
+      }
+    }
+
+    /** Check that this transaction is correctly signed. */
+    def validate(extraUtxos: Map[OutPoint, TxOut]): Boolean = {
+      val inputsMap = extraUtxos + (input.outPoint -> input.txOut)
+      val allInputsProvided = tx.txIn.forall(txIn => inputsMap.contains(txIn.outPoint))
+      val witnessesOk = Try(Transaction.correctlySpends(tx, inputsMap, ScriptFlags.STANDARD_SCRIPT_VERIFY_FLAGS)).isSuccess
+      allInputsProvided && witnessesOk
     }
   }
-  // @formatter:on
 
-  sealed trait ReplaceableTransactionWithInputInfo extends TransactionWithInputInfo {
-    /** Block before which the transaction must be confirmed. */
-    def confirmationTarget: ConfirmationTarget
+  /**
+   * Transactions spending the channel funding output: [[CommitTx]], [[SpliceTx]] and [[ClosingTx]].
+   * Those transactions always require two signatures, one from each channel participant.
+   */
+  sealed trait ChannelSpendTransaction extends TransactionWithInputInfo {
+    /** Sign the channel's 2-of-2 funding output when using a [[SegwitV0CommitmentFormat]]. */
+    def sign(localFundingKey: PrivateKey, remoteFundingPubkey: PublicKey, extraUtxos: Map[OutPoint, TxOut]): ChannelSpendSignature.IndividualSignature = {
+      val redeemScript = Script.write(Scripts.multiSig2of2(localFundingKey.publicKey, remoteFundingPubkey))
+      val sig = sign(localFundingKey, SIGHASH_ALL, RedeemInfo.P2wsh(redeemScript), extraUtxos)
+      ChannelSpendSignature.IndividualSignature(sig)
+    }
+
+    /** Create a partial transaction for the channel's musig2 funding output when using a [[TaprootCommitmentFormat]]. */
+    def partialSign(localFundingKey: PrivateKey, remoteFundingPubkey: PublicKey, extraUtxos: Map[OutPoint, TxOut]): ChannelSpendSignature.PartialSignatureWithNonce = {
```

Comment:
We also need local and remote nonces here:
```suggestion
    def partialSign(localFundingKey: PrivateKey, remoteFundingPubkey: PublicKey, extraUtxos: Map[OutPoint, TxOut], localNonce: (SecretNonce, IndividualNonce), publicNonces: Seq[IndividualNonce]): Either[Throwable, ChannelSpendSignature.PartialSignatureWithNonce] = {
```

We could also simply remove this method and add it in https://github.com/ACINQ/eclair/pull/3079